internal/remoteconfig/remoteconfig_test.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package remoteconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	rc "github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/stretchr/testify/require"
+)
+
+// The RC client relies on Repository (in the datadog-agent) which performs config signature validation
+// using some signing keys which we don't want to expose in this repository, making testing delicate.
+// Testing is performed in the datadog agent for the components being used here, see https://github.com/DataDog/datadog-agent/tree/main/pkg/remoteconfig/state.
+// Signature verification will be changed and made optional in the future, at which point integration testing will become possible
+// as we will be able to setup a Repository and test applying updates, creating a client, etc... all of which require a valid
+// Repository object at the moment
+
+func TestRCClient(t *testing.T) {
+	cfg := DefaultClientConfig()
+	cfg.ServiceName = "test"
+	client := &Client{
+		ClientConfig: cfg,
+		clientID:     generateID(),
+		endpoint:     fmt.Sprintf("http://%s/v0.7/config", cfg.AgentAddr),
+		stop:         make(chan struct{}),
+		callbacks:    map[string][]Callback{},
+	}
+
+	t.Run("registerCallback", func(t *testing.T) {
+		defer func() { client.callbacks = map[string][]Callback{} }()
+		require.Equal(t, 0, len(client.callbacks))
+		client.RegisterCallback(func(ProductUpdate) {}, rc.ProductASMFeatures)
+		require.Equal(t, 1, len(client.callbacks[rc.ProductASMFeatures]))
+		require.Equal(t, 1, len(client.callbacks))
+		client.RegisterCallback(func(ProductUpdate) {}, rc.ProductASMFeatures)
+		require.Equal(t, 2, len(client.callbacks[rc.ProductASMFeatures]))
+		require.Equal(t, 1, len(client.callbacks))
+	})
+
+	/*
+		t.Run("apply-update", func(t *testing.T) {
+			client.Products = append(client.Products, rc.ProductASMFeatures)
+			client.RegisterCallback(func(u ProductUpdate) {
+				require.NotNil(t, u)
+				require.NotNil(t, u[rc.ProductASMFeatures])
+				require.Equal(t, string(u[rc.ProductASMFeatures]), "test")
+			}, rc.ProductASMFeatures)
+
+			client.applyUpdate(&clientGetConfigsResponse{
+				TargetFiles:   []*file{{Path: "path/to/ASM_FEATURES/config", Raw: []byte("test")}},
+				ClientConfigs: []string{rc.ProductASMFeatures},
+			})
+		})
+	*/
```

Comment:
Use `t.Skip()` if you really want to keep this for the future, you should otherwise remove it.

```suggestion
```

Reply:
Nice, thanks for the tip :+1: